src/typechecker.rs

```diff
@@ -3405,30 +3405,31 @@ pub fn typecheck_statement(
 
             (CheckedStatement::While(checked_cond, checked_block), error)
         }
-        ParsedStatement::Return(expr, span) => match expr {
-            Some(e) => {
-                let (output, err) = typecheck_expression(
-                    e,
-                    scope_id,
-                    project,
-                    safety_mode,
-                    project
-                        .current_function_index
-                        .map(|i| project.functions[i].return_type_id),
-                );
-                error = error.or(err);
-
-                if project.inside_defer {
-                    error = error.or(Some(JaktError::TypecheckError(
-                        "‘return’ is not allowed inside ‘defer’".to_string(),
-                        *span,
-                    )));
+        ParsedStatement::Return(expr, span) => {
```

Comment:
This second commit touches the exact same lines as the first one. So, it should be fixed up into the first one.

> Squash your commits when making revisions after a patch review.

Also, the project doesn't use conventional commits, the commit category is supposed to refer to the files, classes, or subsystems you touched in the commit, rather than be `fix: ` or `feat: `

Reply:
I'll get this done and change commit to show the files touched as well